### PR TITLE
[cypress] Fix redirects to the overview page after login

### PR DESCRIPTION
**Pull request by nrfox**
Attempts to fix an issue where kiali will redirect to the overview page after login causing initial page visits during other tests to be replaced with the overview page.

### CODE REVIEW

**Review by jmazzitelli — APPROVED** [review]
After a few failures (due to some local issues I'm having), I was able to get a test run to be successful.

```
 Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  featureFiles/app_details.feature         00:18        7        7        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  featureFiles/apps.feature                00:29       10       10        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  featureFiles/graph_display.feature       00:43       31       31        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  featureFiles/graph_find_hide.featur      00:21        5        5        -        -        - │
  │    e                                                                                           │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  featureFiles/graph_replay.feature        00:16        3        3        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  featureFiles/graph_toolbar.feature       00:24       18       18        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  featureFiles/istio_config.feature        00:21       32       10        -       22        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  featureFiles/kiali_about.feature         00:09        1        1        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  featureFiles/kiali_cookie.feature        00:07        2        2        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  featureFiles/kiali_homepage.feature      00:02        1        1        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  featureFiles/kiali_login.feature         00:03        1        1        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  featureFiles/kiali_logout.feature        00:06        1        1        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  featureFiles/overview.feature            00:40       17       17        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  featureFiles/services.feature            00:28       13       13        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  featureFiles/sidecar_injection.feat      01:02        9        9        -        -        - │
  │    ure                                                                                         │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  featureFiles/wizard_request_routing      00:23        6        6        -        -        - │
  │    .feature                                                                                    │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  featureFiles/workload_logs.feature       00:16        6        6        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  featureFiles/workloads.feature           00:29       11       11        -        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        06:46      174      152        -       22        -  
```

After that, the 22 pending validation tests also passed:

```
    ✓ KIA0101 validation (10998ms)
    ✓ KIA0102 validation (3885ms)
    ✓ KIA0104 validation (3986ms)
    ✓ KIA0106 validation (3790ms)
    ✓ KIA0201 validation (5817ms)
    ✓ KIA0202 validation (3084ms)
    ✓ KIA0203 validation (6038ms)
    ✓ KIA0207 validation (5605ms)
    ✓ KIA0208 validation (5382ms)
    ✓ KIA0209 validation (3584ms)
    ✓ KIA0301 validation (4241ms)
    ✓ KIA0302 validation (2934ms)
    ✓ KIA0505 validation (5310ms)
    ✓ KIA0506 validation (5602ms)
    ✓ KIA1004 validation (3528ms)
    ✓ KIA1006 validation (4608ms)
    ✓ KIA1101 validation (2992ms)
    ✓ KIA1102 validation (4162ms)
    ✓ KIA1104 validation (3631ms)
    ✓ KIA1105 validation (6698ms)
    ✓ KIA1106 validation (5580ms)
    ✓ KIA1107 validation (3741ms)

  22 passing (3m)
```

File: frontend/cypress/support/commands.ts [hunk 1]
```diff
@@ -44,47 +44,49 @@ Cypress.Commands.add('login', (provider: string, username: string, password: str
...
+        // Otherwise the redirect can mess with page loading on subsequent tests.
+        cy.contains('loading', { matchCase: false }).should('not.exist');
+        cy.url().should('include', '/overview');
+        // Wait for the overview page to load some elements to be sure that things
+        // have settled. Otherwise some part of the overview page will change the
+        // url back to the overview page.
+        cy.getBySel('alpha-EXPAND');
       }
+    } else {
+      cy.log('got an auth cookie, skipping login');
```
Comment (ScriptingShrimp) [inline frontend/cypress/support/commands.ts:89]: I need fresh build to verify this on OCP, will take a look tomorrow after fresh nightly. Right now everything fails on this step. 

I do install `hack/istio/install-testing-demos.sh` without any parameters, is that right? @nrfox @jmazzitelli

Comment (jmazzitelli) [reply frontend/cypress/support/commands.ts:89]: Yes. And unfortunately, it seems like for every test run you should delete and then re-install the demos. I've seen errors when I run the cypress tests multiple times without resetting the demos between each run. I haven't figured out why - it is probably due to the fact the cypress tests do not "undo" things they might change during their run. We need these tests to leave the demos in the same state they were in when the tests start, but it seems like at least some tests do not do this.

Use `-d true` to tell the script to uninstall the demos:

```
hack/istio/install-testing-demos.sh -c oc -d true
```

Comment (ScriptingShrimp) [reply frontend/cypress/support/commands.ts:89]: > We need these tests to leave the demos in the same state they were in when the tests start

If this is the case I have some solutions for that

p.s.: install demos script works fine with defaults and fallback to oc 👍

**Pull request merged**